src/main/java/hudson/scm/SubversionSCM.java

```diff
@@ -1371,33 +1369,34 @@ else if (project.getLastBuild()!=null) {
                     return BUILD_NOW;
                 }
                 if (!baseline.revisions.containsKey(url)) {
-                    listener.getLogger().println(
-                            Messages.SubversionSCM_pollChanges_locationNotInWorkspace(url));
+                    listener.getLogger().println(Messages.SubversionSCM_pollChanges_locationNotInWorkspace(url));
                     return BUILD_NOW;
                 }
             }
         }
 
-        // determine where to perform polling. prefer the node where the build happened,
-        // in case a cluster is non-uniform. see http://www.nabble.com/svn-connection-from-slave-only-td24970587.html
-        VirtualChannel ch=null;
+        String nodeName = "master";
+        VirtualChannel channel = null;
         if (workspace != null && !isPollFromMaster()) {
-            ch = workspace.getChannel();
+            channel = workspace.getChannel();
+            if (channel != null && channel instanceof Channel) {
+                nodeName = ((Channel) channel).getName();
+            }
         }
-        if (ch==null)   ch= MasterComputer.localChannel;
 
-        final String nodeName = ch instanceof Channel ? ((Channel) ch).getName() : "master";
+        if (channel == null) {
+            channel = FilePath.localChannel;
+        }
 
         final SVNLogHandler logHandler = new SVNLogHandler(createSVNLogFilter(), listener);
 
-        final Map<String,ISVNAuthenticationProvider> authProviders = new LinkedHashMap<String,
-                ISVNAuthenticationProvider>();
+        final Map<String,ISVNAuthenticationProvider> authProviders = new LinkedHashMap<String, ISVNAuthenticationProvider>();
 
-        EnvVars env = null;
-        Run<?,?> run = project.getLastCompletedBuild();
-        if (run != null) {
-            env = run.getEnvironment(listener);
-        }
+        Node node = Jenkins.getInstance().getNode(nodeName);
+
```

Comment:
I don't think this is correct. `getNode(String)` can never return the master node, so if `isPollFromMaster()` or the last build was on the master node, this should fail. Did you test this at all?